flake8_type_checking/checker.py

```diff
@@ -467,6 +571,24 @@ def __init__(
         self.typing_cast_aliases: set[str] = set()
         self.unquoted_types_in_casts: list[tuple[int, int, str]] = []
 
+    @contextmanager
+    def create_scope(self, node: ast.ClassDef | Function, is_head: bool = True) -> Iterator[Scope]:
+        """Create a new scope."""
+        parent = self.current_scope
+        scope = Scope(node, parent=parent, is_head=is_head)
+        self.scopes.append(scope)
+        self.current_scope = scope
+        yield scope
+        self.current_scope = parent
```

Comment:
It's just to avoid any boilerplate and accidentally forgetting to pop back up to the parent scope, since the AST is traversed depth first you will always enter and exit a scope within the span of a single `visit` call.

Consider this example:
```python
# global scope open
class Foo:  # Foo scope open
    def bar(self):  # bar scope open
        pass # bar & Foo scope close

def baz():  # baz scope open
    pass # baz scope close
```